### ISSUE TITLE
When passing secure connection it's not possible to pass

### ISSUE THREAD

**Issue by m-novikov** [aio-libs/aioredis-py#1209]
### Describe the bug

It not possible to pass custom connection class via kwargs if redis uri starts with `rediss://` (secure connection)

### To Reproduce

```python
from aioredis import connection, from_url


def test_custom_connection_class_connection_pool():
    class MySSLConn(connection.SSLConnection):
        pass

    pool = connection.ConnectionPool.from_url(
        url="rediss://127.0.0.1", connection_class=MySSLConn,
    )
    conn = pool.make_connection()
    assert isinstance(conn, MySSLConn)


def test_from_url_custom_connection_class():
    class MySSLConn(connection.SSLConnection):
        pass

    redis = from_url(
        url="rediss://127.0.0.1", connection_class=MySSLConn,
    )
    conn = redis.connection_pool.make_connection()
    assert isinstance(conn, MySSLConn)
```

### Expected behavior

`connection_class` argument overrides argument derrived from connection string

### Logs/tracebacks

```python-traceback
--
```

### Python Version

```console
$ python --version
3.9
```

### aioredis Version

```console
$ python -m pip show aioredis
Version: 2.0
```

### Additional context

I will followup this issue with a PR

### Code of Conduct

- [X] I agree to follow the aio-libs Code of Conduct

**Comment by Andrew-Chen-Wang** [comment]
tl;dr Justification for why only `rediss` was hard coded: https://github.com/redis/redis-py/pull/446#issuecomment-42906656

---

SSLConnection was implemented in 2014:

- This shows when the class was added: https://github.com/redis/redis-py/commit/de0004b03a04ea2884a06a074a3d382ea8b08a57#diff-45b6ba995a827b484d3ef97cb806d336bf058ff9678b04cf371200ea53e584d0

- This shows when we hard coded SSLConnection to be the conn class given the uri was `rediss` on ConnectionPool creation: https://github.com/redis/redis-py/commit/17ff3774a74a251f17ffb1235b8d38a17fa1b13f#diff-45b6ba995a827b484d3ef97cb806d336bf058ff9678b04cf371200ea53e584d0


This was the reasoning behind it: https://github.com/redis/redis-py/pull/446#issuecomment-42906656

**Comment by Andrew-Chen-Wang** [comment]
Given https://github.com/aio-libs/aioredis-py/issues/1209#issuecomment-974673630, to be as close to redis-py as possible, you should file an issue in the redis-py repo to see if they'd be willing to make the change (especially since it's now maintained under Redis themselves). If they make the change there, we will here. I like Andy's original explanation so I personally would be -1, but again file an issue in redis-py.

**Comment by chayim** [comment]
@Andrew-Chen-Wang In thinking about it - I'm going to merge it in to redis-py.  Generally, I'd like us to start validating against different environment types, and connections - truly testing our SSL, as an example.  

I think the change is a good change - and as I intend to do work within the SSL space in the upcoming month or thereabouts, I think this change is reasonable.

**Comment by Andrew-Chen-Wang** [comment]
Thanks for letting me know @chayim! I'll wait for https://github.com/redis/redis-py/issues/1767 for this change; we're also a little behind on getting on part with redis 5.0.0 😅

**Comment by chayim** [comment]
Happy to help @Andrew-Chen-Wang . We'll be pushing up 4.0.0 rc2 shortly. Ideally tomorrow - but worst case next week.
